src/rgw/rgw_sal_rados.cc

```diff
@@ -990,83 +983,186 @@ int RGWRadosStore::read_all_usage(uint64_t start_epoch, uint64_t end_epoch,
+int RadosStore::trim_all_usage(uint64_t start_epoch, uint64_t end_epoch)
 {
   rgw_user uid;
   std::string bucket_name;
 
   return rados->trim_usage(uid, bucket_name, start_epoch, end_epoch);
 }
 
-int RGWRadosStore::get_config_key_val(string name, bufferlist *bl)
+int RadosStore::get_config_key_val(std::string name, bufferlist* bl)
 {
   return svc()->config_key->get(name, true, bl);
 }
 
-int RGWRadosStore::put_system_obj(const rgw_pool& pool, const string& oid,
-				  bufferlist& data, bool exclusive,
-				  RGWObjVersionTracker *objv_tracker, real_time set_mtime,
-				  optional_yield y, map<string, bufferlist> *pattrs)
+int RadosStore::meta_list_keys_init(const std::string& section, const std::string& marker, void** phandle)
 {
-  auto obj_ctx = svc()->sysobj->init_obj_ctx();
-  return rgw_put_system_obj(obj_ctx, pool, oid, data, exclusive, objv_tracker, set_mtime, y, pattrs);
+  return ctl()->meta.mgr->list_keys_init(section, marker, phandle);
 }
 
-int RGWRadosStore::get_system_obj(const DoutPrefixProvider *dpp,
-				  const rgw_pool& pool, const string& key,
-				  bufferlist& bl,
-				  RGWObjVersionTracker *objv_tracker, real_time *pmtime,
-				  optional_yield y, map<string, bufferlist> *pattrs,
-				  rgw_cache_entry_info *cache_info,
-				  boost::optional<obj_version> refresh_version)
+int RadosStore::meta_list_keys_next(void* handle, int max, list<std::string>& keys, bool* truncated)
 {
-  auto obj_ctx = svc()->sysobj->init_obj_ctx();
-  return rgw_get_system_obj(obj_ctx, pool, key, bl, objv_tracker, pmtime, y, dpp, pattrs, cache_info, refresh_version);
+  return ctl()->meta.mgr->list_keys_next(handle, max, keys, truncated);
 }
 
-int RGWRadosStore::delete_system_obj(const rgw_pool& pool, const string& oid,
-				     RGWObjVersionTracker *objv_tracker, optional_yield y)
+void RadosStore::meta_list_keys_complete(void* handle)
 {
-    return rgw_delete_system_obj(svc()->sysobj, pool, oid, objv_tracker, y);
+  ctl()->meta.mgr->list_keys_complete(handle);
 }
 
-int RGWRadosStore::meta_list_keys_init(const string& section, const string& marker, void** phandle)
+std::string RadosStore::meta_get_marker(void* handle)
 {
-  return ctl()->meta.mgr->list_keys_init(section, marker, phandle);
+  return ctl()->meta.mgr->get_marker(handle);
 }
 
-int RGWRadosStore::meta_list_keys_next(void* handle, int max, list<string>& keys, bool* truncated)
+int RadosStore::meta_remove(const DoutPrefixProvider* dpp, std::string& metadata_key, optional_yield y)
 {
-  return ctl()->meta.mgr->list_keys_next(handle, max, keys, truncated);
+  return ctl()->meta.mgr->remove(metadata_key, y, dpp);
 }
 
-void RGWRadosStore::meta_list_keys_complete(void* handle)
+void RadosStore::finalize(void)
 {
-  ctl()->meta.mgr->list_keys_complete(handle);
+  if (rados)
+    rados->finalize();
 }
 
-std::string RGWRadosStore::meta_get_marker(void* handle)
+std::unique_ptr<LuaScript> RadosStore::get_lua_script()
```

Comment:
name here is a bit misleading. this does not fetch any script, juts the class which is used to manipulate the script or fetch it

Reply:
These getters are named "get_<class_name>, and the class name is LuaScript, so the getter is get_lua_script().  I'm happy to change the name of the class, I wasn't hugely happy with it.

Reply:
this is related to the previous comment. the class should not represent a lua script, juts the mechanism to read/write/delete any lua script. maybe "LuaScriptManager"?

Reply:
Okay, I'll change it.